packages/terra-application-utility/src/menu/_MenuUtilityMenu.jsx

```diff
@@ -0,0 +1,277 @@
+    const childKeys = this.getItem(key).childKeys;
+    const item = this.getItem(key);
+    if (typeof childKeys !== 'undefined' && childKeys.length > 0) {
+      this.toggleIsSelected(key);
+      this.setState({
+        previousKey: this.push(this.state.currentKey),
+        currentKey: key,
+      });
+    } else {
+      this.toggleIsSelected(key);
+      if (item.isSelectable !== true) {
+        this.props.onRequestClose();
+      }
+      this.props.onChange(event, key);
+    }
+  }
+
+  handleRequestBack() {
+    this.setState({ currentKey: this.pop() });
+  }
+
+  pop() {
+    const newStack = this.state.previousKeyStack.slice();
+    const lastKey = newStack.pop();
+    this.setState({ previousKeyStack: newStack });
+    return lastKey;
+  }
+
+  push(key) {
+    const newStack = this.state.previousKeyStack.slice();
+    newStack.push(key);
+    this.setState({ previousKeyStack: newStack });
+  }
+
+  toggleIsSelected(key) {
+    const item = this.getItem(key);
+    item.isSelected = !item.isSelected;
+    this.state.map.set(key, item);
+  }
+
+  render() {
+    const {
+      app,
+      initialSelectedKey,
+      menuItems,
+      isHeightBounded,
+      onChange,
+      onRequestClose,
+      ...customProps
+    } = this.props;
+
+    const MenuClassNames = cx(['utility-menu', customProps.classNames]);
+    const headerClassNames = cx('header');
+    const headerLeftContainer = cx('left-container');
+    const headerRightContainer = cx('right-container');
+    const headerInnerContainer = cx('inner-container');
+    const closebuttonClassNames = cx('close-button');
+    const backButtonClassNames = cx('back-button');
+    const iconLeftClassNames = cx('icon-left');
+    const iconCloseClassNames = cx('icon-close');
+    const headerTextClassNames = cx('header-text');
+    const footerClassNames = cx('footer');
+    const footerDividerClassNames = cx('footer-divider');
+
+    const currentKey = this.state.currentKey;
+    const currentItem = this.getItem(currentKey);
+    const backButton = (
+      <button className={backButtonClassNames} onClick={this.handleRequestBack} aria-label={'Back Button'}>
+        <IconLeft className={iconLeftClassNames} />
+      </button>
+    );
+    const closeButton = (
+      <button onClick={onRequestClose} className={closebuttonClassNames} aria-label={'Close Button'}>
+        <IconClose className={iconCloseClassNames} />
+      </button>
+    );
+
+    const header = (
+      <div className={headerClassNames}>
+        <span className={headerLeftContainer}>
+          <span className={headerInnerContainer}>
+            {currentKey !== Utils.KEYS.MENU && backButton}
+            <span className={headerTextClassNames}>{currentItem.title}</span>
+          </span>
+          <span className={headerRightContainer} >
+            {closeButton}
+          </span>
+        </span>
+        <MenuDivider />
+      </div>
+      );
+
+    let footer = null;
+    if (currentKey === Utils.KEYS.MENU) {
```

Comment:
We shouldn't need to check this, if we have footer items, they should be displayed.